wordpress-mobile/WordPress-Android
# Blaze: Handle click and tracking events on the promote with blaze dashboard card

Parent #17909 

This PR adds the logic for handling event and tracking events for the Promote with Blaze dashboard card.

**To test:**
- Install and launch the app
- Login with a wpcom account 
- ✅ Verify the Promote with Blaze card is not shown
- Navigate to Me > App Settings > Privacy Settings 
- Enable Collect Information
- Navigate to Me > App Settings > Debug Settings
- Enable `blaze`
- Restart the app
- Navigate to the Home dashboard tab
- ✅ Verify the Promote with Blaze card is shown
- ✅ Verify the logs contain:  🔵 Tracked: blaze_feature_displayed, Properties: {"source":"dashboard_card"}
- Tap on the Promote with Blaze Card
- ✅ Verify the logs contain: 🔵 Tracked: blaze_feature_tapped, Properties: {"source":"dashboard_card"}
- Tap on the Promote with Blaze Card more menu (3 vertical dots)
- ✅ Verify the logs contain: 🔵 Tracked: blaze_feature_menu_accessed, Properties: {"source":"dashboard_card"}
- Tap on Promote with Blaze Card > Hide this
- ✅ Verify the logs contain: 🔵 Tracked: blaze_feature_menu_accessed, Properties: {"source":"dashboard_card"}
- ✅ Verify the Promote with Blaze card is not shown
- Swipe close the app
- Reopen the app
- ✅ Verify the Promote with Blaze card is not shown

To reset the card, log out and log in again.

**Note**: The "navigate to blaze" overlay click event will be hooked up following the overlay view implementation. There is a todo in `MySiteViewModel` for completing this step.

## Regression Notes
1. Potential unintended areas of impact
The blaze card is not hidden or shown when it needs to be

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit tests

3. What automated tests I added (or what prevented me from doing so)
Update `MySiteViewModel`

PR submission checklist:
